lib/pf/UnifiedApi/Controller/Redis.pm

```diff
@@ -0,0 +1,78 @@
+package pf::UnifiedApi::Controller::Redis;
+
+=head1 NAME
+
+pf::UnifiedApi::Controller::Redis -
+
+=cut
+
+=head1 DESCRIPTION
+
+pf::UnifiedApi::Controller::Redis
+
+=cut
+
+use strict;
+use warnings;
+use Mojo::Base 'pf::UnifiedApi::Controller::RestRoute';
+use pf::pfqueue::stats;
+
+
+sub queue {
+    my ($self) = @_;
+    my $stats = pf::pfqueue::stats->new;
+    # build hashref
+    my %queue = ();
+    foreach my $item (@{ $stats->queue_counts }) {
+        $queue{$item->{name}} = { count => $item->{count}, outstanding => [], expired => [] };
+    }
+    foreach my $item (@{ $stats->counters }) {
+        push $queue{$item->{queue}}{outstanding}, { name => $item->{name}, count => $item->{count} };
+    }
+    foreach my $item (@{ $stats->miss_counters }) {
+        push $queue{$item->{queue}}{expired}, { name => $item->{name}, count => $item->{count} };
+    }
+    #build json
+    my $json = [];
+    while( my( $key, $value ) = each %queue ){
+        # rebuild hash,
+        #   replace empty []'s w/ undef
```

Comment:
Random question

Why is it better for these to be undef instead of empty arrays ?

Reply:
This is a personal habit for a couple reasons
 * Conditionals are more straight forward on an undef/null than on an empty array
 * It's visually more straight-forward.

Reply:
The thing is its not in our usual practices to do that so this makes it look different than the rest of our codebase

And in the spec we're "promising" an array and this isn't delivering it.

At least to stay consistent with the rest of our code, this should be an empty array

Reply:
Changed in https://github.com/inverse-inc/packetfence/pull/3158/commits/06467b8f7d0f0d0d65ff21b17c4be2072faa16bb